sys/kern/subr_intr.c

```diff
@@ -528,7 +544,11 @@ intr_isrc_deregister(struct intr_irqsrc *isrc)
 		isrc_release_counters(isrc);
 	error = isrc_free_irq(isrc);
 	mtx_unlock(&isrc_table_lock);
-	return (error);
+
+	if (error != 0)
+		return (error);
+
+	return (isrc_event_destroy(isrc));
```

Comment:
As noted above, this routine is never called. Why do we need to call isrc_event_destroy() here? That's new and not quite what the commit message says. How did you test this?

Reply:
Notice how the event is created by `isrc_event_create()`?  Given that, it is INTRNG's job to destroy the event, yet this call was missing.  I simply needed to examine `intr_isrc_deregister()` and notice the missing call.  The driver is sitting in my tree due it needing `intr_isrc_deregister()` to actually work.